iroh-bytes/src/store/file/import_flat_store.rs

```diff
@@ -0,0 +1,387 @@
+//! Import a flat store into the redb store
+//!
+//! This is a separate module since it contains a lot of code that is unrelated
+//! to the rest of the redb store.
+use std::{
+    collections::{BTreeMap, BTreeSet},
+    io,
+    path::{Path, PathBuf},
+};
+
+use crate::{
+    store::{
+        bao_file::raw_outboard_size,
+        file::{tables::Tables, DataLocation, EntryState, OutboardLocation},
+    },
+    util::Tag,
+    IROH_BLOCK_SIZE,
+};
+
+use super::{ActorResult, ActorState, FlatStorePaths};
+use iroh_base::hash::{Hash, HashAndFormat};
+use redb::ReadableTable;
+use std::str::FromStr;
+
+/// A file name that indicates the purpose of the file.
+#[derive(Clone, PartialEq, Eq)]
+pub enum FileName {
+    /// Incomplete data for the hash, with an unique id
+    PartialData(Hash, [u8; 16]),
+    /// File is storing data for the hash
+    Data(Hash),
+    /// File is storing a partial outboard
+    PartialOutboard(Hash, [u8; 16]),
+    /// File is storing an outboard
+    ///
+    /// We can have multiple files with the same outboard, in case the outboard
+    /// does not contain hashes. But we don't store those outboards.
+    Outboard(Hash),
+    #[allow(dead_code)]
```

Comment:
why dead code?

Reply:
This is copied from the old flat store. Not sure. Probably it can be removed.